
Background 
===

Forward genetic screens have been a fundamental strategy to find  genes involved in biological pathways in model species. In these a population is treated with a mutagen that alters the DNA of individuals in some way, e.g. induction of guanine-to-adenine substitutions using ethylmethane sulfonate (EMS) [@Page:2002], then individuals with a phenotype of interest are isolated from a mutagenized population and a recombinant mapping population is created by back-crossing to the parental line or out-crossing to a polymorphic ecotype [@Etherington:2014]. The recombinant population obtained from that cross will segregate for the mutant phenotype and individuals showing the mutant phenotype will carry the causal mutation, even if the genomic location is unknown. The recombination frequency between the causal mutation and nearby genetic markers is low, so the alleles of these linked genetic markers will co-segregate with the phenotype-altering mutation while the remaining unlinked makers segregate randomly in the genome [@Schneeberger:2014aa]. Hence, the allele distribution analysis can unhide these low recombinant regions to identify the location of the causal mutation. This process of genetic analysis is often referred to as bulk segregant analysis (BSA) [@Michelmore:1991aa].

Traditional genetic mapping is a work intensive and time-consuming process but recent advances in high-throughput sequencing (HTS) have greatly accelerated the identification of mutations underlying mutant phenotypes in forward genetic screens.  Several methods as SHOREmap [@Schneeberger:2009], [@Sun:2015], NGM [@Austin:2011] or CandiSNP [@Etherington:2014] based on bulked-segregant analysis of F2 progeny have succeed in the mutant identification in *Arabidopsis thaliana*. All these methods depend on an assembled reference genome and cannot be used in species for which a reference genome is not available. Some alternative solutions as using reference sequences of related species have been proposed [@Wurtzel:2010], [@Livaja:2013aa], but these need low sequence divergence and high levels of synteny between the mutant reads and the related reference sequence and this has restained the application of this approach [@Schneeberger:2014aa],[@Nordstrom:2013aa].

Substantial effort is being made to sequence many species but reasonable completion of a sequence remains expensive and time-comsuming, and fragmented draft genomes present certain limitations in use for mutation mapping in many circumstances. Fast-evolving and repetitive genes such as disease resistance genes [@Song:2003aa] might be absent or divergent from draft reference genome assemblies. Also, draft genomes often have gaps that can frustrate alignments.  

In the last few years, several reference-free methods for general mutation identification have been proposed [Iqbal:2012aa], [Nordstrom:2013aa], [@Minevich:2012aa], [@Abe:2012] to solve the reference sequence restriction, but none have been extended to allow for direct identification of causative mutations.  [@Abe:2012], [@Takagi:2015aa], [@Schneeberger:2014aa]. 

We propose SDM, a fast causative mutant identification method based on a simple reference-free contig assembly that allows the detection of SNPs. Instead of relying on a genome comparison, we focus on the SNP linkage around the causal mutation and analyse the SNP distribution to identify the chromosome area where the putative mutated gene is located. SDM does not rely on previously known genetic markers and it does not need a reference genome. 

