Other information 
==


sup1

(We crossed sup#1 uni-1DT/+ plants with Col wild-type plants, and F2 seeds were obtained from sup#1/+ uni-1DT/+ Ws/Col F1 hybrids. We next collected 88 kanamycin-resistant F2 individuals with wild-type morphologies, 
Because the sup#1 mutation was expected to be located on chr 4, we next examined which region of chr 4 displayed the most significant enrichment of Ws-type homozygous SNPs. To this end, chr 4 was divided into 500 kb intervals and the ratios of homozygous SNPs to heterozygous SNPs in each 500 kb interval were plotted (Fig. 1A). A peak was observed at the 8,000 k–8,500 k interval, suggesting that the sup#1 mutation was located around there. As chr 1 also displayed slight enrichment of homozygous SNPs (Table 2), we drew the same type of graph for chr 1. As expected, a peak was observed in the southern part (Fig. 1B), in agreement with the uni-1DT transgene being located there. Next, to estimate the region containing the sup#1 mutation more precisely, the 6,000 k–10,000 k region of chr 4 was divided into 50 kb intervals and enrichment of homozygous SNPs in each 50 kb interval was plotted (Fig. 1C). Relatively high peaks were detected between 6,850 k and 9,100 k, and thus we decided to focus on the 6,800 k–9,150 k region that contains all high peaks.)




--

The development of de novo assembly technologies based on PacBio or Illumina long-read sequencing have reduced the computational time required to obtain a draft genome and allow the obtention of data to close gaps and address low quality regions in existing genome assemblies [@Koren:2015aa], [@Kim:2014aa]. N50 contig length may reach several kilobases and this will also end with the need of mapping to a reference genome to identify mutated genes. 

--


![](http://www.sciweavers.org/tex2img.php?eq=Deviation%20%3D%20%20%5Cfrac%7B%20%20%7C%20candidate-causal%20%7C%20%20%7D%7BLength%7Dx100&bc=White&fc=Black&im=jpg&fs=9&ff=arev&edit=0)








(For simplicity, the 151 candidate positions predicted for sup#1 were not itemised here.) 




![equation](http://www.sciweavers.org/tex2img.php?eq=Ratio=\frac{Hom%2B1}{het%2B1}&bc=White&fc=Black&im=jpg&fs=9&ff=arev&edit=0)
