
#Identification of genomic regions carrying a causal mutation in unordered genomes.

##Abstract
##Background
- NGS useful for mappping mutations.
- Forward genetic screens.
- SNP/Variant calling methods require a reference genome: NGM, SHOREmap, CandiSNP, MutMap... Advantages and limitations
- Reference-free methods. NIKS, De Bruijn graphs. Limitations
- Brief explanation of SDM. 
Unfinished assemblies can complicate practical application of SHOREmapping.  **Limited access to the reference genome sequence reduces the number of markers that can be accessed, thus limiting the power for identifying causal regions.**

##Results

######1. SDM design and modelling using Arabidopsis chromosome I and ideallized SNP densities
######2. SDM detects the high SNP density peak with a negligible deviation from the expected peak when using model genomes 
######3. A pre-filtering step based on homozygous to heterozygous SNPs ratio improves SDM accuracy
######4. Confirmation of the causative mutations identified by bulk segregant analysis using Illumina sequencing and Arabidopsis as a reference genome 
######5.Homozygous SNPs in forward genetics screens are normally distributed around the causative mutation
######6. SDM accuracy improvement by filtering background SNPs and centromeres
######7. SDM identified the genomic region carrying the causal mutation previously found with other methods

##Discussion
- backcross vs outcross experiments
- Advantages and future perspectives

SDM performance and comparison to other methods



