#References
1. Page DR, Grossniklaus U: The art and design of genetic screens: Arabidopsis thaliana. Nat Rev Genet 2002, 3:124–36.

2. Etherington GJ, Monaghan J, Zipfel C, MacLean D: Mapping mutations in plant genomes with the user-friendly web application candiSNP. Plant Methods 2014, 10:41.

3. Schneeberger K: Using next-generation sequencing to isolate mutant genes from forward genetic screens. Nat Rev Genet 2014, 15:662–76.

4. Michelmore RW, Paran I, Kesseli RV: Identification of markers linked to disease-resistance genes by bulked segregant analysis: A rapid method to detect markers in specific genomic regions by using segregating populations. Proc Natl Acad Sci U S A 1991, 88:9828–32.

5. Schneeberger K, Ossowski S, Lanz C, Juul T, Petersen AH, Nielsen KL, Jørgensen J-E, Weigel D, Andersen SU: SHOREmap: Simultaneous mapping and mutation identification by deep sequencing. Nat Methods 2009, 6:550–1.

6. Sun H, Schneeberger K: SHOREmap v3.0: Fast and accurate identification of causal mutations from forward genetic screens. Methods Mol Biol 2015, 1284:381–95.

7. Austin RS, Vidaurre D, Stamatiou G, Breit R, Provart NJ, Bonetta D, Zhang J, Fung P, Gong Y, Wang PW, McCourt P, Guttman DS: Next-generation mapping of arabidopsis genes. Plant J 2011, 67:715–25.

8. Wurtzel O, Dori-Bachash M, Pietrokovski S, Jurkevitch E, Sorek R: Mutation detection with next-generation resequencing through a mediator genome. PLoS One 2010, 5:e15628.

9. Livaja M, Wang Y, Wieckhorst S, Haseneyer G, Seidel M, Hahn V, Knapp SJ, Taudien S, Schön C-C, Bauer E: BSTA: A targeted approach combines bulked segregant analysis with next- generation sequencing and de novo transcriptome assembly for sNP discovery in sunflower. BMC Genomics 2013, 14:628.

10. Nordström KJV, Albani MC, James GV, Gutjahr C, Hartwig B, Turck F, Paszkowski U, Coupland G, Schneeberger K: Mutation identification by direct comparison of whole-genome sequencing data from mutant and wild-type individuals using k-mers. Nat Biotechnol 2013, 31:325–30.

11. Song J, Bradeen JM, Naess SK, Raasch JA, Wielgus SM, Haberlach GT, Liu J, Kuang H, Austin-Phillips S, Buell CR, Helgeson JP, Jiang J: Gene rB cloned from solanum bulbocastanum confers broad spectrum resistance to potato late blight. Proc Natl Acad Sci U S A 2003, 100:9128–33.

12. Fonseca VG, Carvalho GR, Sung W, Johnson HF, Power DM, Neill SP, Packer M, Blaxter ML, Lambshead PJD, Thomas WK, Creer S: Second-generation environmental sequencing unmasks marine metazoan biodiversity. Nat Commun 2010, 1:98.

13. Koren S, Phillippy AM: One chromosome, one contig: Complete microbial genomes from long-read sequencing and assembly. Curr Opin Microbiol 2015, 23:110–20.

14. Kim KE, Peluso P, Babayan P, Yeadon PJ, Yu C, Fisher WW, Chin C-S, Rapicavoli NA, Rank DR, Li J, Catcheside DEA, Celniker SE, Phillippy AM, Bergman CM, Landolin JM: Long-read, whole-genome shotgun sequence data for five model organisms. Sci Data 2014, 1:140045.

15. Minevich G, Park DS, Blankenberg D, Poole RJ, Hobert O: CloudMap: A cloud-based pipeline for analysis of mutant genome sequences. Genetics 2012, 192:1249–69.

16. Abe A, Kosugi S, Yoshida K, Natsume S, Takagi H, Kanzaki H, Matsumura H, Yoshida K, Mitsuoka C, Tamiru M, Innan H, Cano L, Kamoun S, Terauchi R: Genome sequencing reveals agronomically important loci in rice using mutMap. Nat Biotechnol 2012, 30:174–8.

17. Takagi H, Tamiru M, Abe A, Yoshida K, Uemura A, Yaegashi H, Obara T, Oikawa K, Utsushi H, Kanzaki E, Mitsuoka C, Natsume S, Kosugi S, Kanzaki H, Matsumura H, Urasaki N, Kamoun S, Terauchi R: MutMap accelerates breeding of a salt-tolerant rice cultivar. Nat Biotechnol 2015, 33:445–9.

18. Galvão VC, Nordström KJV, Lanz C, Sulz P, Mathieu J, Posé D, Schmid M, Weigel D, Schneeberger K: Synteny-based mapping-by-sequencing enabled by targeted enrichment. Plant J 2012, 71:517–26.

19. Allen RS, Nakasugi K, Doran RL, Millar AA, Waterhouse PM: Facile mutant identification via a single parental backcross method and application of whole genome sequencing based mapping pipelines. Front Plant Sci 2013, 4:362.

20. Monaghan J, Matschi S, Shorinola O, Rovenich H, Matei A, Segonzac C, Malinovsky FG, Rathjen JP, MacLean D, Romeis T, Zipfel C: The calcium-dependent protein kinase cPK28 buffers plant immunity and regulates bIK1 turnover. Cell Host Microbe 2014, 16:605–15.

21. Uchida N, Sakamoto T, Tasaka M, Kurata T: Identification of eMS-induced causal mutations in arabidopsis thaliana by next-generation sequencing. Methods Mol Biol 2014, 1062:259–70.

22. Bolger AM, Lohse M, Usadel B: Trimmomatic: A flexible trimmer for illumina sequence data. Bioinformatics 2014, 30:2114–20.

23. Li H, Durbin R: Fast and accurate long-read alignment with burrows-wheeler transform. Bioinformatics 2010, 26:589–95.

24. Koboldt DC, Zhang Q, Larson DE, Shen D, McLellan MD, Lin L, Miller CA, Mardis ER, Ding L, Wilson RK: VarScan 2: Somatic mutation and copy number alteration discovery in cancer by exome sequencing. Genome Res 2012, 22:568–76.

25. Koboldt DC, Chen K, Wylie T, Larson DE, McLellan MD, Mardis ER, Weinstock GM, Wilson RK, Ding L: VarScan: Variant detection in massively parallel sequencing of individual and pooled samples. Bioinformatics 2009, 25:2283–5.

26. Team RDC: R: A Language and Environment for Statistical Computing. Vienna, Austria: R Foundation for Statistical Computing; 2011.

27. Komsta L, Novomestky F: Moments, Cumulants, Skewness, Kurtosis and Related Tests. 2015.

28. Lamesch P, Berardini TZ, Li D, Swarbreck D, Wilks C, Sasidharan R, Muller R, Dreher K, Alexander DL, Garcia-Hernandez M, Karthikeyan AS, Lee CH, Nelson WD, Ploetz L, Singh S, Wensel A, Huala E: The arabidopsis information resource (tAIR): Improved gene annotation and new tools. Nucleic Acids Res 2012, 40(Database issue):D1202–10.

29. DePristo MA, Banks E, Poplin R, Garimella KV, Maguire JR, Hartl C, Philippakis AA, Angel G del, Rivas MA, Hanna M, McKenna A, Fennell TJ, Kernytsky AM, Sivachenko AY, Cibulskis K, Gabriel SB, Altshuler D, Daly MJ: A framework for variation discovery and genotyping using next-generation dNA sequencing data. Nat Genet 2011, 43:491–8.

30. Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R, 1000 Genome Project Data Processing Subgroup: The sequence alignment/Map format and sAMtools. Bioinformatics 2009, 25:2078–9.

31. Melters DP, Bradnam KR, Young HA, Telis N, May MR, Ruby JG, Sebra R, Peluso P, Eid J, Rank D, Garcia JF, DeRisi JL, Smith T, Tobias C, Ross-Ibarra J, Korf I, Chan SWL: Comparative analysis of tandem repeats from hundreds of species reveals unique insights into centromere evolution. Genome Biol 2013, 14:R10.