######SDM can find the high SNP density peak when using Arabidopsis thaliana chromosome I and ideallized SNP densities as a model


dedicated code hosting website GitHub at https://github.com/pilarcormo/SNP_distribution_method



######Homozygous SNPs in forward genetics screens are normally distributed around the causative mutation

Unfinished assemblies can complicate practical application of SHOREmapping.  **Limited access to the reference genome sequence reduces the number of markers that can be accessed, thus limiting the power for identifying causal regions.**

We selected different examples of bulk segregant analysis of a mutation segregating in an outcross or backcross population. We performed convencional genome alignment to the reference genome using the reads provided in 4 different mapping-by-sequencing experiments ([@Galvao:2012], [@Allen:2013], [@Monaghan:2014], [Uchida:2014]]) that succeed in identifying the mutation. The techniques used to identify the mutations were different in every case so we took advantage of this fact to confirm the reproducibility of the methods available. To reproduce the experiments, we aligned the Illuminapaired-end reads to the Arabidopsis thaliana reference genome (TAIR10) after a quality filtering step. Then, we used VarScan for variant detection. After filtering the parental SNPs and removing the variability due to the centomere (Figure 1.A), we plotted the SNP densitiy. Then, we analysed the homozygous SNPs distribution in each chromosome and the ratio homozygous to heterozygous to confirm the genomic region where the mutations were described. SNPs with a frequency of 0.8 or larger were considered homozygous. 

In the first study, they generated a recombinant mapping population by outcrossing a Columbia (Col-0) background mutant to the divergent accession Landsberg erecta (Ler). The offspring were allowed to self-fertilize. A mutant pool of 119 F2 individuals was sequenced (we refer to this dataset as **OCF2**) [@Galvao:2012]. The two parental lines Col and Ler were also sequenced. Then, they performed conventional SHOREmap [@Schneeberger:2009] to identify the mutation causing late flowering which lied on chromosome 2, specifically on gene _SOC1_(18807538..18811047).

In an attempt to identify the gene involved in the Arabidopsis microRNA pathway, the mutants in the second study (**BCF2** dataset)[@Allen:2013] were generated by back-crossing a mutant to the non-mutagenized parental line followed by one round of self-fertilization. A pool of 110 mutant individuals showing leaf hyponasty and the non-mutagenized parental line mir159a were sequenced. They used different SNP callers that produced highly similar outcomes (NGM, SHOREmap, GATK and samtools)[@Austin:2011], [@Schneeberger:2009], [@DePristo:2011aa], [@Li:2009aa]. They identified the causal SNP mutation in _HASTY_, a gene on chromosome 3 (1401271..1408197). 

We also analysed a recent forward screen done in the immune-deficient bak1-5 background to identify novel components involved in plant immune signalling [@Monaghan:2014]. In this screen, after mutagenizing bak1-5 seeds (in the Col-0 ecotype) with EMS, they screened the M2 generation for modifier of **bak1-5 (mob) mutants** that restored immune signalling. To obtain the mutants, they back-crossed bak1-5 mob mutants to the parental line (bak1-5). F2 mutants and the parental line (bak1-5) were sequenced. They used CandiSNP [@Etherington:2014] to map the mutations and identified causal mutations in the gene encoding the calcium-dependent protein kinase CPK28 (26456285..26459631) for both bak1-5 mob1 and bak1-5 mob2.

The last mapping-by-sequencing study we used was obtained by outcrossing a Arabidopsis Wassilewskija (Ws) background mutant to Col wild-type plants followed by sequecing of 88 F2 individuals and Ws and Col as parental lines [Uchida:2014]. They described a pipeline to identify the causal mutation based on the peaks obtained by plotting ratios of homozygous SNPs to heterozygous SNPs. By following this approach, they reported that the majority of SNPs detected on chr 4 were homozygous and they identified the sup#1 mutation on the _SGT1b_ gene (6851277..6853860). They also shown a homozygous peak on chromosome 1 due to the original transgene being located there. 

(We crossed sup#1 uni-1DT/+ plants with Col wild-type plants, and F2 seeds were obtained from sup#1/+ uni-1DT/+ Ws/Col F1 hybrids. We next collected 88 kanamycin-resistant F2 individuals with wild-type morphologies, 
Because the sup#1 mutation was expected to be located on chr 4, we next examined which region of chr 4 displayed the most significant enrichment of Ws-type homozygous SNPs. To this end, chr 4 was divided into 500 kb intervals and the ratios of homozygous SNPs to heterozygous SNPs in each 500 kb interval were plotted (Fig. 1A). A peak was observed at the 8,000 k–8,500 k interval, suggesting that the sup#1 mutation was located around there. As chr 1 also displayed slight enrichment of homozygous SNPs (Table 2), we drew the same type of graph for chr 1. As expected, a peak was observed in the southern part (Fig. 1B), in agreement with the uni-1DT transgene being located there. Next, to estimate the region containing the sup#1 mutation more precisely, the 6,000 k–10,000 k region of chr 4 was divided into 50 kb intervals and enrichment of homozygous SNPs in each 50 kb interval was plotted (Fig. 1C). Relatively high peaks were detected between 6,850 k and 9,100 k, and thus we decided to focus on the 6,800 k–9,150 k region that contains all high peaks.)

Once we obtained the homozygous SNP densities after running the standarized variant calling approach and the filtering steps, we proceed to plot the homozygous and heterozygous SNP densities. For all the forward genetics screens under analysis we confirmed that there was a high homozygous SNP density in the genomic area where the causal SNP was previous described, suggesting a linkage of homozygous SNPs to the phenotype-altering SNP in a low recombination area. By plotting the densities in all the chromosomes after filtering, we could confirm that the high SNP density peak was unique for all the cases apart from one. The outcrossed sup#1 mutant also showed a peak in chromosome 1, but this was expected and described in the original report [Uchida:2014]. We divide the chromosomes in 100 kb fragments and calculate the ratios of homozygous SNPs to heterozygous SNPs as described by the formula in figure 1. We overlapped both homozuygous and heterozygous densities and the ratio. The result is (**figure 2.A**). 

To determine if the homozygous SNP distribution obtained correlated with a theoretical probability distribution, we built probability plots or Q-Q plots (**figure 2.B**). Our results indicate that homozygous SNP densities in bulked segregant analysis in Arabidopsis correlate to a normal distribution with a variable standard deviation that oscillates between 3 and 7 Mb. We also analysed the shape of the distributions by measuring the kurtosis. 4 of the 5 distributions were platykurtic, as they showed a negative kurtosis and consequently a lower, wider peak around the mean and thinner tails. Only one (sup#1) was leptokurtic (positive kurtosis). We also measured the skewness; this value implies that the distribution of the data is skewed to the left (or negatively skewed) or to the right (positively skewed). The skewness will depend on the location of the causal mutation in the chromosome, so as expected, we did not see any conserved tendency among the data (**Table 1**). 


<table>
  <tr><th>Sample <th>Number of homozygous SNPs</th><th>Chromosome</th><th>Correlation with normal distribution (r<sup>2</sup>)</th><th>Standard deviation (Mb)</th><th>Kurtosis</th><th>Skewness</th>
  
  <tr><th> OCF2</th> <th>151</th> <th>2</th><th>>0.9<th>6.01</th><th>1.85</th><th>-0.392</th>
  
  <tr><th> BCF2</th>  <th>15</th><th>3</th><th>>0.9<th>3.20</th><th>2.02</th><th>0.201</th>
  
  
  <tr><th> bak1-5 mob1</th>  <th>25</th><th> 5</th><th>>0.9<th>7.20 </th><th>2.69</th><th>-0.240</th>
 
  
  <tr><th>bak1-5 mob2 </th> <th>41</th> <th> 5</th><th>~0.9<th>3.71</th><th>2.37</th><th>0.445</th>  
   
   
  <tr><th> sup#1</th>  <th>4633</th><th>4</th><th>>0.95<th>3.66</th><th>3.50</th><th>0.370</th>
  
</table>

Figure 2. 4 plots with all the mutations identified and qqplots.

######SDM accuracy improvement by filtering background SNPs 