Results and discussion
===

######SDM detects the high SNP density peak with a negligible deviation from the expected peak when using model genomes 

To develop our method to identify an experimentally induced causal mutation, we created different model genomes (**Additional fig. 1**) based on _Arabidopsis thaliana_ chromosome. We tested different parameters that could have a potential effect on SDM performance: genome length, contig size, homozygous SNP density, heterozygous SNP density, way of calculating the homozygous to heterozygous ratio and location of the causal mutation.

We used a set of unordered contigs and the ideal SNP distribution as starting point. SDM sorts the fragments by their SNP density values so that they follow a Gaussian distribution. The homozygous SNPs to heterozygous SNPs (Hom/het) ratio is also measured. Due to the SNP linkage to the causal mutation, the ratio will  be maximum in the area where the causative SNP is located. We analysed the effect of the factor added to numerator and denominator (**Additional fig. 2**). We meaused the ratio by using different values (1, 0.1 and 0.01) but did not find any conclusive improvement with any of them so decided to use 1. 

To estimate the effect of the genome length, we created model genomes of different sizes ranging from 1 Mb to 15 Mb. To test the influence of the contig size, we used a variable contig length randomly ranging between a provided mininum value and a maximum value, calculated by doubling the minimum. We used two minimum contig sizes (500 bp and 1000 bp) that gave rise to approximately 1300 and 700 contigs respectively. 

We started using a high SNP density (1 SNP each 500 bp), with the causal mutation located in the middle of the distribution. By using the model, we define beforehand the location of the  causative mutation so we can measure the deviation (**Fig. 1A**) of SDM from the expected position. To guarantee the confidence of SDM, we created 5 replicates for each condition and calculated the average deviation from the causal mutation for all the genome sizes and contig sizes (**Fig. 1B**). We proved that SDM identified the high density peak with no significant effect of genome length and contig size. The deviation from the causal mutation assigned in the model was always lower than 0.5%, so we can conclude that our method was reliable to estimate the mutation under these conditions.  We repeated the previous experiment by using the whole chromosome I and 3 different contig sizes. The results were similar to those for smaller genomes, and again we did not see any significant effect of genome length or contig size (**Fig. 1C**). 

######A pre-filtering step based on the homozygous to heterozygous SNPs ratio improves SDM accuracy 

SDM worked perfectly when the idealised causal mutation was located in the middle of the distribution (mean) as the number of fragments at both sides (right and left) was the same. When we shifted the causal mutation in our model to one side (one tail was longer than the other), SDM was not able to sort the contigs properly. The reason for this shift is that  contigs with the same SNP frequency are impossible to classify as belonging to the right or left tail. 

We delimited a threshold value based on the ratio of homozygous to heterozygous SNPs. The contigs with a ratio below the threshold are not considered in the analysis, and only those with a high ratio are sorted. Even though the exact position in the genome cannot be determined by this approach, we can assess the contigs in which the mutation is more likely to be found, dismissing a great part of the genome that is hiding the causative mutation. To test the influence of the threshold we used the whole chromosome I from *Arabdidopsis* as a model. We tested different ratio values from 0.5 to 2 (**Fig. 2**). 0.5 or 1 were discriminative enough for SDM to find the correct order of the relevant contigs. We standardised this value and used a 25% of the maximum ratio as threshold.


######Confirmation of the causative mutations identified by bulk segregant analysis using Illumina sequencing and Arabidopsis as a reference genome 

We selected different sets of data of bulk segregant analysis of a mutation segregating in an out-cross [@Galvao:2012], [@Uchida:2014]] or back-cross [@Allen:2013], [@Monaghan:2014] population. We performed convencional genome alignment to the reference genome using the reads provided in the 4 mapping-by-sequencing experiments. The techniques used to identify the mutations were different in every case so we took advantage of this fact to confirm the reproducibility of the available methodology.  We aligned the Illumina paired-end reads to the *Arabidopsis thaliana* reference genome. Then, we used VarScan for variant calling. After plotting the SNP densitiy (**Additional file 4**), we analysed the homozygous SNPs distribution in each chromosome and the ratio of homozygous to heterozygous SNPs to confirm the genomic region where the mutations were described. 

In the first out-cross experiment we analised (OCF2) [@Galv√£o et al. 2012] they identified the mutation causing late flowering which lied on chromosome 2, specifically on gene _SOC1_(18807538..18811047). In the second study (BCF2) they analised the mutant individuals showing leaf hyponasty to identify a gene involved in the Arabidopsis microRNA pathway. They identified the causal SNP mutation in _HASTY_, a gene on chromosome 3 (1401271..1408197). In a recent forward screen done in the immune-deficient bak1-5 background to identify novel components involved in plant immune signalling [@Monaghan:2014], they identified causal mutations in the gene encoding the calcium-dependent protein kinase CPK28 (26456285..26459631) for both bak1-5 mob1 and bak1-5 mob2. In the last mapping-by-sequencing study we used they reported that the majority of SNPs detected on chromosome 4 were homozygous and they identified the sup#1 mutation on the _SGT1b_ gene (6851277..6853860). They also shown a homozygous peak on chromosome 1 due to the transgene present in the original population. 

Once we obtained the homozygous SNP densities, we proceed to plot the homozygous and heterozygous SNP densities. For all the forward genetics screens under analysis we confirmed that there was a high homozygous SNP density peak in the genomic area where the causal SNP was previously described (**Fig. 3**). These results corroborate the idea of linkage of homozygous SNPs to the phenotype-altering SNP in the region where recombination events are unlikely. 

The high homozygous density peaks were unique in the genome after filtering in all the studies except for the out-crossed sup#1 mutant,  where a high density peak was also found in chromosome I. This was expected and described in the original report [@Uchida:2014].

######Homozygous SNPs in forward genetics screens are normally distributed around the causative mutation 

After running the standardised variant calling approach with the different data sets, we showed a unique peak in the SNP distribution around the causative mutation (**Fig. 3**). The next step was to analyse the correlation of the SNP distribution to a theoretical probability distribution. For this aim, we built probability plots or Q-Q plots with the homozygous SNP densities in the back-cross and out-cross experiments (**Fig. 4A**). Our results indicate that there is a good correlation between the homozygous SNP frequencies and a normal distribution. We further validated the correlation between the sample values and the predicted values by a simple linear regression (r<sup>2</sup> >0.9). The standard deviation, although variable between samples, oscillated between 3 and 7 Mb (**Table 1**). 

We also analysed the shape of the distributions by measuring the kurtosis (**Table 1**). 4 of the 5 distributions were platykurtic, as they showed a negative kurtosis and consequently a lower, wider peak around the mean and thinner tails. Only one (sup#1) was leptokurtic (positive kurtosis). We also measured the skewness; this value implies that the distribution of the data is skewed to the left (negatively skewed) or to the right (positively skewed). The skewness will depend on the location of the causal mutation in the chromosome, so as expected, we did not see any conserved tendency among the data (**Table 1**). 

######Filtering background SNPs and centromeres unmasks the high homozygous SNP density peak 

We filtered the background SNPs to reduce the complexity of the distribution and improve the degree of correlation to a bell curve. The parental reads were subjected to the same variant calling approach and those SNPs that were present in both parental and mutant samples were removed from the mutant SNP list (**Fig. 3B**). The usefulness of removing non-unique SNPs is not new, and all the  mapping-by-sequencing  experiments we have chosen here did the same filtering to some extent. ([@Galvao:2012], [@Uchida:2014]], [@Allen:2013], [@Monaghan:2014])

SDM accuracy was improved when the we removed SNPs concentrated around centromeres (**Additional fig. 3**). The use of *Arabidopsis thaliana* reads simplified this filtering step since centromeres are well defined in the genome assembly. F or the long-term goal of the methodology proposed, we still have to develop some way to identify the variability due to the centromere. Many studies have shown the peculiarity of the centromere, so these sequences should be easy to identify. Centromeres are  characterised by  the presence of tandem repeats, and extremely high repeat abundance (often >10,000 copies per chromosome) [@Melters:2013aa].

######SDM identifies the genomic region carrying the causal mutation previously described by other methods

As a final proof-of-concept, we used the SNP densities obtained from OCF2, BCF2, mob1-mob2 and sup#1 datasets to create new model genomes. We took the chromosomes where the mutations were described and split them into fragments. 

We regained the normal distribution for all the datasets after shuffling the contig order and running SDM (**Fig. 6A**). The correct contig order and real ratio distribution are known, so we created a plot comparing the hypothetical ratio distribution after SDM and the real one to test the performance of SDM (**Fig. 6B**). 

In our case studies, the causal mutation estimation by SDM was more accurate in back-crosses (bak1-5 mutants and BCF2) than in out-crosses (OCF2 and sup#1). The number of segregating SNPs is lower in back-cross experiments (1 SNP every 65,000 bp) compared to out-crosses (1 SNP every 900 bp) [@Etherington:2014]. The large number of segregating SNPs in out-crosses complicated the assignation of a position in the normal distribution to a contig. For example, if more than 2 contigs contain 4 SNPs, there is no reliable way to discriminate between them and the problem gets worse when the number of contigs with the same number of SNPs increases. Hence, we added a feature to SDM to specify the type of cross before defining the candidate mutations. 

As expected, the approximation to the causal mutation is more accurate in back-cross experiments and the number of candidates oscillates between 4 and 10 (**Table 2**). In our two out-cross examples the number of homozygous SNPs differs; SDM identified the causal mutation position in OCF2 (**Table 2**) even though the number of candidates was greater compared to the back-crosses. However, more than 100 candidate mutations out of several thousand SNPs were proposed for sup#1 in a very small region of the chromosome (100 kb) and the actual causal SNP was not in that list (**Table 2**). Therefore, out-crosses to a reference line would need more filtering steps to reduce the number of SNPs at the starting point as they did in the original report [Uchida:2014]. We can conclude that SDM is a rapid and precise method to perform bulk segregant linkage analysis from back-crossed populations without relying on the disponibility of a reference genome, but we need to be cautious when dealing with out-cross experiments.  


Conclusions
===

Forward genetic screens are very useful to identify genes responsible for particular phenotypes. Thanks to the advances in HTS technologies, mutant genomes sequencing has become quick and unexpensive. However, the mapping-by-sequencing methods available present certain limitations, complicating the mutation identification especially in non-sequenced species. To target this problem, we proposed a fast, reference genome independent method to identify causative mutations. We showed that homozygous SNPs are normally distributed in the mutant genome of back-cross and out-crossed individuals. Based on that idea, we defined a theoretical SNP distribution used by SDM to identify the genomic region where the causative mutation was located. We conclude that SDM is especially sucessful for analysing mutants obtained from a back-cross population. The increase in the number of SNPs in out-cross experiments complicated the genetic analysis and the mutation estimation. Ideally, over the next few years, sequencing costs will decrease and this will allow to sequence every mutant individual from a forward genetic screen. Therefore, we need fast and reliable methods to identify variants bypassing the reference genome assembly step. We now aim to improve and apply SDM in forward genetics screens of species where a reference genome is not yet available. Also, we plan to develop an accessible software that will speed up gene finding in non-sequenced organisms. 


