3. Results and discussion
===

###3.1. SDM works over a range of effective genome lengths and realistic fragment sizes

We created model genomes based on _Arabidopsis thaliana_ chromosomes to develop our mutant identification method. We chose *Arabidopsis thaliana* to create our model genomes because it is been a widely used organism for forward genetic screening and it is a relatively small and well-annotated genome. Also, SNP densities in several mapping-by-sequencing experiments in Arabidopsis are available (see section 3.3 for several examples) so they could be used as a basis to develop our methodology. 

By making customised genomes, we could rapidly alter different paramenters as genome length, contig size or SNP density to analyse their effect on the detection method accuracy. Our dinamic way of creating model genomes helped us define all the different aspects that should be taken into account when analysing the SNP distribution. Also, the causative mutation was defined manually by us, so we could measure the deviation between defined and predicted value. We generated the model genomes by asigning an idealised SNP distribution to a set of randomly shuffled sequences that imitate contigs assembled from HTS. In the model genomes, homozygous SNPs follow a normal distribution while heterozygous SNPs follow a uniform distribution. 

To estimate the effect of the genome length on SDM's ability to find the causative mutation, we created model genomes of different sizes ranging from 1 Mb to 15 Mb. We defined a variable contig length randomly ranging between a provided mininum value and its double. We used two minimum contig sizes (500 bp and 1000 bp) that gave rise to approximately 1300 and 700 contigs respectively. To guarantee the confidence of SDM, we created 5 replicates for each condition. 
For the second set of model genomes, we used the whole chromosome I length and a more realistic SNP density (1 SNP every 3000 bp). In this case, 3 contig sizes were employed (1000, 2000 and 4000 approximately).

The SNP Distribution Method (SDM) sorts the sequence fragments by their SNP density values (the score calculated as defined in section 2.2) so that they follow a normal distribution. Then, a second sorting step in which the contigs are also sorted by their ratio (calculated as defined in section 2.2) is applied. We considered the highest kernel density value for the SNP distribution after SDM sorting as the candidate SNP location and this value was compared to the peak of the normal distribution previously defined in the model genome. The difference was called 'deviation' and can be found in [**Fig.2**](Fig2/deviation_stata.png). Consistent results were obtained for all the replicates. SDM identified the high density peak with no significant effect of genome length and contig size. The deviation from the causative mutation assigned in the model was lower than 1% in the small SNP-riched genomes (Fig 2A). This was also true for the whole-sized genomes when they were divided in 1000 and 2000 contigs but not for the 4000 contigs (Fig 2B), where 4 replicates had a deviation between 1 and 3%. Even though the deviation is still low, the higher the number of contigs, the harder it is to get to the correct order. Also, as we did not change the SNP density, the SNPs are spreaded over more fragments and the sorting is more complicated. 


###3.2. A pre-filtering step based on the homozygous to heterozygous SNPs ratio improves SDM accuracy 

As explained in the previous section, SDM succeed in the high density peak identification in model genomes when the idealised causative mutation was located in the middle of the distribution as the number of fragments at both sides (right and left) was the same. However, this success is only true in the high SNP density area (peak of the distribution) and when we shifted the causal mutation in our model to one side (one tail was longer than the other), SDM was not able to sort the contigs in the tails properly. Even though the contigs located in the peak were those in which the mutation was defined, the algorithm was not able to classify the low density contigs as belonging to the right or left tail and the distribution peak appeared moved, giving rise to a high deviation from the defined peak. 

To fix the problem, we delimited a threshold value based on the ratio of homozygous to heterozygous SNPs to discard contigs located further away from the causative mutation. We excluded those contigs with a ratio below a given percentage (1%, 5%, 10% or 20%) of the maximum ratio. Therefore, only those contigs with a high ratio are sorted. Even though the exact position in the genome cannot be determined by this approach, we can assess the contigs in which the mutation is more likely to be found, dismissing a great part of the genome that is hiding the causative mutation. To test the influence of the threshold we used the whole chromosome I from *Arabdidopsis* as a model genome [(**Fig. 3**)](Fig3/ratios_5percent.png). The high density peak in the ratio distribution matched the expected when aproximately a 80-90% of the genome is discarded. In this case, the SNP density was high (1 SNP every 3000 bp) and the distribution peak was narrow. However, we cannot establish a standard threshold, since it will depend on the peak deviation on each case. To avoid lossing interesting mutations by being too strict with the filtering, we suggest using 1-5% of the maximum ratio to obtain optimal results. 

 
###3.3. Filtering background SNPs and centromeres unmasks the high homozygous SNP density peak in several bulk segregant analysis in Arabidopsis 

We selected different sets of data of bulk segregant analysis of a mutation segregating in an out-cross [@Galvao:2012], [@Uchida:2014] or back-cross [@Allen:2013], [@Monaghan:2014] population. We performed conventional genome alignment to the reference genome using the reads provided in the 4 studies. The techniques used to identify the mutations ([**Table 1**](tables.pdf)) were different in every case so we took advantage of this fact to confirm the reproducibility of the available methodology to identify causative mutations.  We aligned the Illumina paired-end reads to the *Arabidopsis thaliana* reference genome. Then, we used VarScan for variant calling. 

In the first out-cross experiment we analysed (OCF2), GalvÃ£o et al identified the mutation causing late flowering which lied on chromosome 2, specifically on gene _SOC1_ (18807538..18811047) **[@Galvao. 2012]** . In the second study (BCF2) Allen at al analised the mutant individuals showing leaf hyponasty to identify a gene involved in the Arabidopsis microRNA pathway [@Allen et al. 2013]. They identified the causal SNP mutation in _HASTY_, a gene on chromosome 3 (1401271..1408197). Also, we used the reads from a forward screen done in the immune-deficient bak1-5 background aim identify new components involved in plant immunity. Monaghan et al found 2 causative mutations in the gene encoding the calcium-dependent protein kinase CPK28 (26456285..26459631) for both bak1-5 mob1 and bak1-5 mob2 [@Monaghan:2014]. In the last mapping-by-sequencing study we used, Uchida et al reported that the majority of SNPs detected on chromosome 4 were homozygous and they identified the sup#1 mutation on the _SGT1b_ gene (6851277..6853860) [@Uchida:2014] ([**Table 2**](tables.pdf))

The main advantage of working with already identified mutations was to focus directly on the chromosome were the mutation was previously described. We analysed the total number of homozygous SNPs in the chromosome where the causative mutation was located. As expected, when the mutant individual is out-crossed to a distant mapping line (OCF2 and sup#1), the SNP density is up to 20 times higher than in the case of back-crossing to the parental line (BCF2 and mob mutants). In the back-crossed populations, we identified approximately 1700 homozygous SNPs. That gives an overall density of 1 SNP every 15000 bp. We identified 9200 homozygoys SNPs in the chromosome of interest in the first out-crossed population (OCF2) and 27000 SNPs in the second out-crossed population (sup#1). The overall density was of 1 SNP every 2500 bp for OCF2 and 1 SNP every 700 bp for sup#1. 

Parental filtering was fundamental to reduce the SNP density and unmask the SNP linkage around the causative mutation and it was especially crucial in out-crossed populations where the starting density was higher. The parental lines used to back or out-cross were also sequenced and mapped to the Arabidopsis thaliana reference genome. The SNPs present in the non-mutant parental reads were onsidered as 'background' mutations and  filtered from the mutant SNP lists. In the back-cross studies, the absolute homozygous SNP number was reduced up to 9 times ([**Fig4A**](Fig4/densities.png)) after parental filtering.  The total number of homozygous SNPs was reduced 3 times in out-crossed populations  ([**Fig4C**](Fig4/densities.png)). Even though the centromere removal did not reduce the total number of SNPs in the same proportion as parental filtering did, it was essential to unmask the normal distribution around the causative mutation. Many studies have shown the centromeres peculiarity, characterised by high repeat abundance (often >10,000 copies per chromosome) [@Melters:2013aa]. This high variability in a few hundred of bp generates a high SNP density peak which hides the peak of interest. 

The filtering steps reduced the complexity of the distribution ([Additional fig 3](Additional/addit3.png)) and improved the degree of correlation to a normal distribution (see section 3.5). The usefulness of removing non-unique SNPs is not new, and all the  mapping-by-sequencing  experiments we used did the same filtering to some extent. ([@Galvao:2012], [@Uchida:2014]], [@Allen:2013], [@Monaghan:2014])

We identified a unique peak in the area where the causative mutation was described when we plotted the homozygous SNP density obtained after filtering. We calculated the homozygous to heterozygous ratio for each contig and the ratio values were overlapped to the SNP densities, obtaining the distributions in [**Fig 5**](Fig5/densities_good.png). All the densities were congruent with the results described in the publications.  

###3.4. Homozygous SNPs in forward genetics screens are normally distributed around the causative mutation 

After running the variant calling approach with the different datasets, we showed a unique peak in the SNP distribution around the causative mutation (**Fig. 3**). The next step was to analyse the correlation of the SNP distribution to a theoretical probability distribution. We created probability plots or Q-Q plots with the homozygous SNP densities in the back-crossed and out-crossed populations ([**Fig 6**](Fig6/qqplots.png)). Our results indicate a good correlation between the homozygous SNP frequencies and a normal distribution. We further validated the correlation between the sample values and the predicted values by a simple linear regression (r<sup>2</sup> >0.9). The standard deviation, although variable between samples, oscillated between 3 and 7 Mb ([**Table 2**](tables.pdf)). Therefore, this is evidence that 15 Mb model genomes are large enough to identify the normal density peak when using SNP densities from real forward genetic screens. 

We also analysed the shape of the distributions by measuring the kurtosis ([**Table 3**](tables.pdf)). 4 of the 5 distributions were platykurtic, as they showed a negative kurtosis and consequently a lower, wider peak around the mean and thinner tails. Only one (sup#1) was leptokurtic (positive kurtosis). This might be due to the high density observed in sup#1 (more than 3 times higher than the other out-crossed population) which generates a narrower peak due to the greater linkage of SNPs in the non-recombinant area. In the other examples, the number of SNPs linked together is lower and they are highly scattered in the genome, generating the lower and wider peak. 


###3.5. The N50 contig size in plant genome assemblies depends on genome size and sequencing technology

We planed to generate more realistic model genomes. To decide on a representational contig size, 29 assemblies at contig level were analysed. The relationship between genome length and N50 contig size was not really obvious as other aspects as the sequencing technology used **([Fig7A](Fig7/contigs.png))** or the genome coverage have a high impact on the final N50 contig size. We observed that the preferred sequencing technology in the last 2-3 years is Illumina HiSeq since half of the assemblies were built using this technology. Other assemblies used combinations of different techniques (including Illumina, 454 and PacBio). The median value of the N50 contig length for all the 29 assemblies is 11,517 bp while it is reduced to 5,484 bp when analysing only HiSeq assemblies **([Fig7B](Fig7/contigs.png))**. To decrease the effect of the technology used, we focused on the 16 assemblies built with Illumina Hiseq and we tried to establish a model that could explain the N50 size change over genome size.  

The relationship was not linear but we did not have any mechanistic model to describe it, so we apply a Generalized Additive Model (GAM) to fit non-parametric smoothers to the data without specifing a particular model. First, we applied logarithms to both N50 size and genome length on the Illumina HiSeq assemblies. When we focused on those genomes larger than 200 Mb, the R square was 0.807 and p-value was 0.000909.

We used 3 different contig sizes to create the model genomes. The first two model genomes were built using the N50 median values. We chose 5,000 bp (based on the median for Illumina Hiseq) and 10,000 bp (based on the median for all the assemblies). The last contig size was decided looking at the model defined for Illumina HiSeq. Due to the *Arabidopsis thaliana* genome size, the minimum contig size decided for these model genomes was 2,000 bp.

###3.6. SDM identifies the genomic region carrying the causal mutation previously described by other methods

As a final proof-of-concept, we used the SNP densities obtained from OCF2, BCF2, mob1, mob2 and sup#1 datasets as explained in section 3.3 after parental filtering and centromere removal to build new model genomes. We split the *Arabidopsis thaliana* chromosomes where the mutations into fragments as described in section 3.5.

We regained the normal distribution for all the datasets after shuffling the contig order and running SDM. We created a plot to compare the hypothetical ratio distribution after SDM and the real one to test the performance of SDM. The results for all the model genomes generated were deposited at a Github repository at [https://github.com/pilarcormo/SNP_distribution_method/tree/master/arabidopsis_datasets/No_centromere](https://github.com/pilarcormo/SNP_distribution_method/tree/master/arabidopsis_datasets/No_centromere).

Our prior knowledge about the correct contig order allowed us to define the real chromosomic positions in the artificially-made contigs identified by SDM as candidates. In that way, we were able to adjust the method to get its best efficiency ([**Table 4**](tables.pdf)). 

Contig size had an effect on the number of candidate contigs supplied by SDM. When the minimum contig sizes were 2 and 5 kb, the SNP positions were split to different contigs and it was more difficult for SDM to find the correct contig order. When the average contig size was below 10 kb, 20 candidate contigs were considered for back-crossed populations and 40 were needed for out-crossed populations due to the high SNP densities. However, when the average contig size was over 10 kb, 12 candidate contigs in the middle of the distribution were enough to contain the causative mutation. The candidate region ranges from 60 to 180 kb depending on the contig size and the type of cross. 

In our example, the number of candidate positions was higher for the out-crosses than for the back-crosses. This is logical as the SNP density after filtering was higher in the out-crossed populations, and therefore, the number of segerating SNPs in the candidate contigs is larger. 

We could not define a universal cut-off value based on the homozygous to heterozygous ratio for all the different SNP densities, as sometimes the region with a high ratio was narrow due to a high SNP linkage in the area, while in other cases, the increase in the ratio was progressive, and the peak was wider. Therefore, when we worked with real densities, we used an automatic approach that tailor the threshold for each specific SNP density anc contig length. [**Table 4**](tables.pdf) shows the tailored thresholds and total discarded contigs for all the datsets.

We can conclude that SDM is a rapid and precise method to perform bulk segregant linkage analysis from back-crossed and out-crossed populations without relying on the disponibility of a reference genome, specially effective on large contig sizes/ 
 
Conclusions
===

Forward genetic screens are very useful to identify genes responsible for particular phenotypes. Thanks to the advances in HTS technologies, mutant genomes sequencing has become quick and unexpensive. However, the mapping-by-sequencing methods available present certain limitations, complicating the mutation identification especially in non-sequenced species. To target this problem, we proposed a fast, reference genome independent method to identify causative mutations. We showed that homozygous SNPs are normally distributed in the mutant genome of back-cross and out-crossed individuals. Based on that idea, we defined a theoretical SNP distribution used by SDM to identify the genomic region where the causative mutation was located. We conclude that SDM is especially sucessful for analysing mutants obtained from a back-cross population. The increase in the number of SNPs in out-cross experiments complicated the genetic analysis and the mutation estimation. Ideally, over the next few years, sequencing costs will decrease and this will allow to sequence every mutant individual from a forward genetic screen. Therefore, we need fast and reliable methods to identify variants bypassing the reference genome assembly step. We now aim to improve and apply SDM in forward genetics screens of species where a reference genome is not yet available. We plan to develop an accessible software that will speed up gene finding in non-sequenced organisms. 


